### PR TITLE
[803] detect_device function simplified

### PR DESCRIPTION
**Pull request by turtledreams**

### CODE REVIEW

**Comment by turtledreams** [comment]
cant recall is_bot, ua_raw thingy, lets chat again

**Comment by turtledreams** [comment]
made it so that user agent related information can be processed from a single place and accessed from there. However still looks messy, might assign each value to their own function

**Review by ArtursKadikis — CHANGES_REQUESTED** [review]
divide this into 3 different calls

**Comment by turtledreams** [comment]
current tests:

image

File: lib/countly.js [hunk 1]
```diff
@@ -4138,37 +4139,52 @@
...
+        return uaOverride || navigator.userAgent;
+    }
+
+    /**
+     *  Returns device type information according to user agent string
+     *  @memberof Countly._internals
+     *  @param {string} uaOverride - a string value to pass instead of ua value
+     *  @returns {string} detect_device - current device type (desktop, tablet, phone)
+     */
+    function detect_device(uaOverride) {
```
Comment (turtledreams) [inline lib/countly.js:4158]: indicate user agent is used

File: lib/countly.js [hunk 2]
```diff
@@ -4138,37 +4139,52 @@
...
+        return device;
+    }
+
+    /**
+     *  Returns information regarding if the current user is a search bot or not
+     *  @memberof Countly._internals
+     *  @param {string} uaOverride - a string value to pass instead of ua value
+     *  @returns {boolean} is_bot - if a search bot is reaching the site or not
+     */
+    function is_bot(uaOverride) {
```
Comment (turtledreams) [inline lib/countly.js:4182]: user agent is used

**Review by ArtursKadikis — APPROVED** [review]

**Pull request merged**